twisted/internet/defer.py

```diff
@@ -661,6 +661,66 @@ def __str__(self):
     __repr__ = __str__
 
 
+    def __await__(self):
+        awaiter = _MakeDeferredAwaitable(self)
```

Comment:
It's just that `__next__` and `__iter__` will be in C. Also, `YIELD_FROM` opcode has a fast-path for generator objects. This is still a micro-optimization, but you might want to write a micro-benchmark.